pennylane/ops/qubit.py

```diff
@@ -1136,13 +1156,8 @@ def diagonalizing_gates(self):
         Returns:
             list: list containing the gates diagonalizing the Hermitian observable
         """
-        Hmat = self.matrix
-        Hkey = tuple(Hmat.flatten().tolist())
-        if Hkey not in Hermitian._eigs:
-            w, U = np.linalg.eigh(Hmat)
-            Hermitian._eigs[Hkey] = {"eigval": w, "eigvec": U}
         return [
-            QubitUnitary(Hermitian._eigs[Hkey]["eigvec"].conj().T, wires=self.wires),
+            QubitUnitary(self.eigendecomposition["eigvec"].conj().T, wires=list(sorted(self.wires))),
```

Comment:
Sure that the wires must be sorted here? The matrix diagonalized by `self.eigendecomposition["eigvec"].conj().T` is also meant for the wires in unsorted order

Reply:
The input matrix of ``Hermitian`` has been permutated, before the decomposition has been computed (see the def of ``Hermitian.eigendecomposition()``).

Steps that may improve the fact that sorting seems to happen out of nowhere:
*  Rename ``Hermitian.eigendecomposition()`` such that it implies that the decomposition was computed on a permuted matrix (e.g.  ``Hermitian.eigendecomposition_of_permuted()`` or something similar)
*  Create a  ``Hermitian._permuted_wires`` property which does the sorting and might improve readability (although might still not provide too much explanation in this mere code snippet)